### PR TITLE
nsinit: Add a flag to enable systemd support for cgroups

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by rjnagal** [comment]
LGTM

File: nsinit/utils.go [hunk 1]
```diff
@@ -29,7 +30,11 @@ func loadConfig(context *cli.Context) (*configs.Config, error) {
 }
 
 func loadFactory(context *cli.Context) (libcontainer.Factory, error) {
-	return libcontainer.New(context.GlobalString("root"), libcontainer.Cgroupfs)
+	cgm := libcontainer.Cgroupfs
+	if context.Bool("systemd") && systemd.UseSystemd() {
```
Comment (LK4D4) [inline nsinit/utils.go]: Shouldn't we log warning if systemd isn't available, but flag is supplied?

Comment (mrunalp) [inline nsinit/utils.go]: Done.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**